packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts

```diff
@@ -0,0 +1,228 @@
+import { useInfiniteQuery, useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+import Thread from 'models/Thread';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import {
+  ThreadTimelineFilterTypes
+} from '../../../models/types';
+import { cacheTypes } from './helpers/cache';
+
+const THREADS_STALE_TIME = 180000; // 3 minutes
+
+const QueryTypes = {
+  ACTIVE: cacheTypes.ACTIVE_THREADS,
+  BULK: cacheTypes.BULK_THREADS
+}
+
+const queryTypeToRQMap = {
+  bulk: useInfiniteQuery,
+  active: useQuery,
+}
+
+interface CommonProps {
+  queryType: typeof QueryTypes[keyof typeof QueryTypes],
+  chainId: string;
+}
+
+interface FetchBulkThreadsProps extends CommonProps {
+  queryType: typeof QueryTypes.BULK; // discriminating union
+  page: number;
+  limit?: number;
+  toDate: string;
+  fromDate?: string;
+  topicId?: number;
+  stage?: string;
+  includePinnedThreads?: boolean;
+  orderBy?: 'newest' | 'oldest' | 'mostLikes' | 'mostComments'
+}
+
+interface FetchActiveThreadsProps extends CommonProps {
+  queryType: typeof QueryTypes.ACTIVE; // discriminating union
+  topicsPerThread?: number;
+}
+
+const featuredFilterQueryMap = {
+  newest: 'createdAt:desc',
+  oldest: 'createdAt:asc',
+  mostLikes: 'numberOfLikes:desc',
+  mostComments: 'numberOfComments:desc',
+};
+
+const useDateCursor = ({ dateRange }: { dateRange?: ThreadTimelineFilterTypes }) => {
+  const [dateCursor, setDateCursor] = useState<{
+    toDate: string,
+    fromDate: string | null
+  }>({ toDate: moment().toISOString(), fromDate: null })
+
+  useEffect(() => {
+    const today = moment();
+
+    const updater = () => {
+      const fromDate = (() => {
+        if (dateRange) {
+          if (
+            [
+              ThreadTimelineFilterTypes.ThisMonth,
+              ThreadTimelineFilterTypes.ThisWeek,
+            ].includes(dateRange)
+          ) {
+            return today
+              .startOf(dateRange.toLowerCase().replace('this', '') as any)
+              .toISOString();
+          }
+
+          if (dateRange.toLowerCase() === ThreadTimelineFilterTypes.AllTime) {
+            return new Date(0).toISOString();
+          }
+        }
+
+        return null;
+      })();
+      const toDate = (() => {
+        if (dateRange) {
+          if (
+            [
+              ThreadTimelineFilterTypes.ThisMonth,
+              ThreadTimelineFilterTypes.ThisWeek,
+            ].includes(dateRange)
+          ) {
+            return today
+              .endOf(dateRange.toLowerCase().replace('this', '') as any)
+              .toISOString();
+          }
+
+          if (dateRange.toLowerCase() === ThreadTimelineFilterTypes.AllTime) {
+            return moment().toISOString();
+          }
+        }
+
+        return moment().toISOString();
+      })();
+      setDateCursor({ toDate, fromDate })
+    }
```

Comment:
please move `fromDate` and `toDate` to util functions, probably helpers directory is a good choice as well.

Reply:
I agree with the thinking but in this case, it might lead to unexpected behavior (maybe later down the down if someone unknowingly updates that helper). PS The logic is only used here so maybe we can util'ize (make it a util) in the same file?

Reply:
oh do you mean `/threads/helper` ?

Reply:
yes threads/helper

Reply:
Done